### PR TITLE
#709 extract mempool service

### PR DESCRIPTION
**Pull request by rozhkovdmitrii**

### CODE REVIEW

File: proxy/common_neon/utils/queue_based_service.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            pass
+
+        MemPoolQueueManager.register('get_queue')
+        queue_manager = MemPoolQueueManager(address=(host, port), authkey=b'abracadabra')
+        queue_manager.connect()
+        self._queue = queue_manager.get_queue()
+
+    def invoke(self, method_name, *args, **kwargs):
+        invocation = ServiceInvocation(method_name=method_name, args=args, kwargs=kwargs)
+
```
Comment (rozhkovdmitrii) [inline proxy/common_neon/utils/queue_based_service.py:42]: extra space

File: proxy/common_neon/utils/queue_based_service.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        class MemPoolQueueManager(BaseManager):
+            pass
+
+        MemPoolQueueManager.register("get_queue", callable=lambda: self._queue)
+        self._queue_manager = MemPoolQueueManager(address=('', port), authkey=b'abracadabra')
+        self._mempool_server = self._queue_manager.get_server()
+        self._mempool_server_process = mp.Process(target=self._mempool_server.serve_forever, name="mempool_service")
+        self._queue_process = mp.Process(target=self.run, name="mempool_queue_proc")
+        self._timeout = self.QUEUE_TIMEOUT_SEC
```
Comment (rozhkovdmitrii) [inline proxy/common_neon/utils/queue_based_service.py]: move up

File: proxy/common_neon/utils/queue_based_service.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        if not self._is_back_ground:
+            self._queue_process.join()
+
+    def run(self):
+        self.service_process_init()
+        while True:
+            try:
+                if not self._run_impl():
+                    break
+            except BaseException as e:
```
Comment (rozhkovdmitrii) [inline proxy/common_neon/utils/queue_based_service.py]: queue Empty

File: proxy/common_neon/utils/queue_based_service.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                self.do_extras()
+
+    def _run_impl(self) -> bool:
+
```
Comment (rozhkovdmitrii) [inline proxy/common_neon/utils/queue_based_service.py]: extra space

File: proxy/mempool/mempool_client.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+from . import MemPoolService
+
+
+@logged_group("neon")
+class MemPoolClient(QueueBasedServiceClient):
+
+    MEM_POOL_SERVICE_HOST = "127.0.0.1"
+
+    def __init__(self):
+        self.info("Construct MemPoolClient")
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mempool_client.py]: initialize

File: proxy/mempool/mempool_service.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+from logged_groups import logged_group
+
+from ..common_neon.utils import QueueBasedService
+
+from .mem_pool import MemPool
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mempool_service.py:5]: from .

File: proxy/mempool/mem_pool.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
+from logged_groups import logged_group
+from multiprocessing import Pool
+
+
+@logged_group("neon.Proxy")
```
Comment (afalaleev) [inline proxy/mempool/mem_pool.py]: I think better to add a new logged_group 'neon.Mempool'.
This will show next stages in transaction processing.

**Review by afalaleev — APPROVED** [review]

**Review by otselnik — APPROVED** [review]

File: proxy/mempool/mem_pool.py (hunk 7 above)
Comment (rozhkovdmitrii) [reply proxy/mempool/mem_pool.py]: done

**Pull request merged**